dlp/src/main/java/com/example/dlp/RiskAnalysis.java

```diff
@@ -53,90 +64,135 @@
+
+      // create /action to publish job status notifications over Google Cloud Pub/Sub
+      Action action = Action.newBuilder().setPubSub(publishToPubSub).build();
+
+      RiskAnalysisJobConfig riskAnalysisJobConfig =
+          RiskAnalysisJobConfig.newBuilder()
               .setSourceTable(bigQueryTable)
+              .setPrivacyMetric(privacyMetric)
+              .addActions(action)
               .build();
 
-      // asynchronously submit a risk analysis operation
-      OperationFuture<RiskAnalysisOperationResult, RiskAnalysisOperationMetadata>
-          responseFuture = dlpServiceClient.analyzeDataSourceRiskAsync(request);
+      CreateDlpJobRequest createDlpJobRequest =
+          CreateDlpJobRequest.newBuilder()
+              .setParent(ProjectName.of(projectId).toString())
+              .setRiskJob(riskAnalysisJobConfig)
+              .build();
 
-      // ...
-      // block on response
-      RiskAnalysisOperationResult response = responseFuture.get();
-      NumericalStatsResult results =
-          response.getNumericalStatsResult();
+      DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+      String dlpJobName = dlpJob.getName();
 
-      System.out.println(
-          "Value range: [" + results.getMaxValue() + ", " + results.getMinValue() + "]");
+      // wait on job completion
+      waitOnJobCompletion(projectId, subscriptionId, dlpJobName);
 
-      // Print out unique quantiles
-      String previousValue = "";
-      for (int i = 0; i < results.getQuantileValuesCount(); i++) {
-        Value valueObj = results.getQuantileValues(i);
-        String value = valueObj.toString();
+      // retrieve completed job status
+      DlpJob completedJob =
+          dlpServiceClient.getDlpJob(GetDlpJobRequest.newBuilder().setName(dlpJobName).build());
 
-        if (!previousValue.equals(value)) {
-          System.out.println("Value at " + i + "% quantile: " + value.toString());
-          previousValue = value;
-        }
+      System.out.println("Job status: " + completedJob.getState());
+      AnalyzeDataSourceRiskDetails riskDetails = completedJob.getRiskDetails();
+      AnalyzeDataSourceRiskDetails.NumericalStatsResult result =
+          riskDetails.getNumericalStatsResult();
+
+      System.out.printf(
+          "Value range : [%.3f, %.3f]\n",
+          result.getMinValue().getFloatValue(), result.getMaxValue().getFloatValue());
+
+      int percent = 1;
+      for (Value quantileValue : result.getQuantileValuesList()) {
+        System.out.printf(
+            "Value at %s %% quantile : %.3f", percent, quantileValue.getFloatValue());
       }
-    } catch (Exception e) {
-      System.out.println("Error in numericalStatsAnalysis: " + e.getMessage());
     }
-    // [END dlp_numerical_stats]
   }
 
-  private static void calculateCategoricalStats(
-      String projectId, String datasetId, String tableId, String columnName)
+  // [START wait_on_dlp_job_completion]
```

Comment:
Nit: should we keep this method in a {single, separate} file/location, instead of including it in multiple files?

Reply:
Removed this method, so this comment is no longer relevant